addons/binding/org.openhab.binding.irtrans/src/main/java/org/openhab/binding/irtrans/handler/EthernetBridgeHandler.java

```diff
@@ -0,0 +1,1024 @@
+                    new Object[] { ipAddress, portNumber, e.getMessage() });
+        }
+
+        while (selector == null) {
+            try {
+                Thread.sleep(100);
+            } catch (InterruptedException e) {
+                logger.error("An exception occurred while putting the Thread to sleep: {}", e.getMessage(), e);
+            }
+        }
+
+        synchronized (selector) {
+            selector.wakeup();
+            int interestSet = SelectionKey.OP_READ | SelectionKey.OP_WRITE | SelectionKey.OP_CONNECT;
+            try {
+                socketChannelKey = socketChannel.register(selector, interestSet);
+            } catch (ClosedChannelException e1) {
+                logger.error("An exception occurred while registering a selector: {}", e1.getMessage());
+            }
+        }
+
+        InetSocketAddress remoteAddress = new InetSocketAddress(ipAddress, portNumber);
+
+        try {
+            logger.trace("Connecting the channel for {} ", remoteAddress);
+            socketChannel.connect(remoteAddress);
+        } catch (Exception e) {
+            logger.error("An exception occurred while connecting connecting to '{}:{}' : {}",
+                    new Object[] { ipAddress, portNumber, e.getMessage() });
+        }
+
+        return socketChannel;
+    }
+
+    private void disconnect(SocketChannel socketChannel) {
+        logger.trace("Disconnecting the socket channel '{}'", socketChannel);
+        try {
+            socketChannel.close();
+        } catch (IOException e) {
+            logger.warn("An exception occurred while closing the channel '{}': {}", socketChannel, e.getMessage());
+        }
+    }
+
+    protected void onAcceptable(ServerSocketChannel listenerChannel) {
+        lock.lock();
+        try {
+            synchronized (selector) {
+                try {
+                    selector.selectNow();
+                } catch (IOException e) {
+                    logger.error("An exception occurred while selecting: {}", e.getMessage());
+                }
+            }
+
+            Iterator<SelectionKey> it = selector.selectedKeys().iterator();
+            while (it.hasNext()) {
+                SelectionKey selKey = it.next();
+                it.remove();
+                if (selKey.isValid()) {
+                    if (selKey.isAcceptable() && selKey == listenerChannel.keyFor(selector)) {
+                        try {
+                            SocketChannel newChannel = listenerChannel.accept();
+                            logger.trace("Received a connection request from '{}'", newChannel.getRemoteAddress());
+
+                            try {
+                                newChannel.configureBlocking(false);
+                            } catch (IOException e) {
+                                logger.error("An exception occurred while configuring the channel '{}': {}", newChannel,
```

Comment:
Same here, imho it's unlikely that you can continue like this if the statement on line 515 failed. Note that if you would leave out the try catch here, it would be caught on line 530 which looks way more correct.